engine/table/src/main/java/io/deephaven/engine/table/impl/sources/RowKeyColumnSource.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.engine.table.impl.sources;
+
+import io.deephaven.chunk.LongChunk;
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.chunk.WritableLongChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.rowset.chunkattributes.RowKeys;
+import io.deephaven.engine.table.impl.AbstractColumnSource;
+import io.deephaven.engine.table.impl.ImmutableColumnSourceGetDefaults;
+import io.deephaven.util.QueryConstants;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * This is a column source that uses no additional memory and is an identity mapping from row key to row key.
+ */
+public class RowKeyColumnSource extends AbstractColumnSource<Long>
+        implements ImmutableColumnSourceGetDefaults.ForLong, FillUnordered<Values> {
+    public static final RowKeyColumnSource INSTANCE = new RowKeyColumnSource();
+
+    public RowKeyColumnSource() {
+        super(Long.class);
+    }
+
+    @Override
+    public long getLong(long rowKey) {
+        return rowKey < 0 ? QueryConstants.NULL_LONG : rowKey;
+    }
+
+    @Override
+    public void fillChunk(
+            @NotNull final FillContext context,
+            @NotNull final WritableChunk<? super Values> destination,
+            @NotNull final RowSequence rowSequence) {
+        doFillChunk(destination, rowSequence);
+    }
+
+    @Override
+    public void fillPrevChunk(
+            @NotNull final FillContext context,
+            @NotNull final WritableChunk<? super Values> destination,
+            @NotNull final RowSequence rowSequence) {
+        doFillChunk(destination, rowSequence);
+    }
+
+    public static void doFillChunk(
+            @NotNull final WritableChunk<? super Values> destination,
+            @NotNull final RowSequence rowSequence) {
+        final WritableLongChunk<? super Values> longChunk = destination.asWritableLongChunk();
+        if (rowSequence.isContiguous()) {
+            final int size = rowSequence.intSize();
+            longChunk.setSize(size);
+            final long firstRowKey = rowSequence.firstRowKey();
+            for (int ii = 0; ii < size; ++ii) {
+                longChunk.set(ii, firstRowKey + ii);
+            }
+        } else {
+            rowSequence.fillRowKeyChunk(longChunk);
```

Comment:
I'm wondering if I was wrong to suggest the complexity and we should just use this.

Reply:
As discussed, the contiguous optimization seemed to be ~8% improvement in a small scenario with ~100k symbols. The performance measurement was not localized to the CS implementation, so it seems significant enough to keep. Noting that I did add a check to see if I should fill by row key ranges and that dropped back down to never having the contiguous optimization. It's possible it's all a wash, but this impl consistently came out on top.